common/src/tap_manager.rs

```diff
@@ -86,14 +86,20 @@ impl TapManager {
         // TODO: consider doing this in another async task to avoid slowing down the paid query flow.
         sqlx::query!(
             r#"
-                INSERT INTO scalar_tap_receipts (allocation_id, timestamp_ns, receipt)
-                VALUES ($1, $2, $3)
+                INSERT INTO scalar_tap_receipts (allocation_id, sender_address, timestamp_ns, value, receipt)
+                VALUES ($1, $2, $3, $4, $5)
             "#,
             format!("{:?}", allocation_id)
                 .strip_prefix("0x")
                 .unwrap()
                 .to_owned(),
+                receipt_signer
+                .to_string()
+                .strip_prefix("0x")
+                .unwrap()
+                .to_owned(),
```

Comment:
In both these cases it would be better to use [`trim_start_matches("0x")`](https://doc.rust-lang.org/std/string/struct.String.html#method.trim_start_matches). `strip_prefix` returns `None` if the allocation ID is lacking a `0x` prefix. If we're doing validations anywhere, we should do it in a way that reports an error back to the client and does not crash the service.

Reply:
Ah, seems I didn't RTFM hard enough! Yes that'd be much better indeed, thanks!

Reply:
[f7447f7](https://github.com/graphops/indexer-service-rs/pull/83/commits/f7447f7f27a8436b74b4d20ce7abb878e576c354)